pandas/tests/base/test_unique.py

```diff
@@ -105,3 +105,24 @@ def test_nunique_null(null_obj, index_or_series_obj):
         num_unique_values = len(obj.unique())
         assert obj.nunique() == max(0, num_unique_values - 1)
         assert obj.nunique(dropna=False) == max(0, num_unique_values)
+
+
+@pytest.mark.parametrize(
+    "idx_or_series_w_bad_unicode", [pd.Index(["\ud83d"]), pd.Series(["\ud83d"])]
+)
+def test_unique_bad_unicode(idx_or_series_w_bad_unicode):
+    # regression test for #34550
+    obj = idx_or_series_w_bad_unicode
+    obj = np.repeat(obj, range(1, len(obj) + 1))
+    result = obj.unique()
+
+    # dict.fromkeys preserves the order
+    unique_values = list(dict.fromkeys(obj.values))
+    if isinstance(obj, pd.Index):
+        expected = pd.Index(unique_values, dtype=obj.dtype)
+        if is_datetime64tz_dtype(obj.dtype):
+            expected = expected.normalize()
+        tm.assert_index_equal(result, expected)
+    else:
+        expected = np.array(unique_values, dtype=obj.dtype)
+        tm.assert_numpy_array_equal(result, expected)
```

Comment:
I think this could be simpler. Since the sole unique value is hard-coded above we can just explicitly construct the expected output using that value and object dtype (shouldn't need to check for datetime).

Reply:
Thanks for your comment @dsaxton.  I actually wanted to replicate the basic test earlier in the same file, `test_unique(..)`.  It takes a fixture, and I wasn't sure how to pass a fixture and my examples simultaneously to `pytest.mark.parametrize` (my search led me to a feature request for `pytest` with lots of messy workarounds).  So I guess the options are:

1. I simplify this as you suggest, or
2. we figure out a way to pass these examples along with the fixture.

This is my first PR here, so a bit of guidance would be great :)

Reply:
Sure thing, testing PRs are always appreciated. It's probably fine to create this as a simple one-off rather than including invalid unicode in every test that uses the fixture.

Reply:
Okay, I'll simplify this one and push an update.  Thanks a lot :)